shared/js/ui/base/ui-wrapper.es6.js

```diff
@@ -9,54 +9,67 @@ const sendMessage = async (messageType, options) => {
 const backgroundMessage = (thisModel) => {
     // listen for messages from background and
     // // notify subscribers
-    window.chrome.runtime.onMessage.addListener((req, sender) => {
-        if (sender.id !== chrome.runtime.id) return
+    browser.runtime.onMessage.addListener((req, sender) => {
+        if (sender.id !== browser.runtime.id) return
         if (req.allowlistChanged) thisModel.send('allowlistChanged')
         if (req.updateTabData) thisModel.send('updateTabData')
         if (req.didResetTrackersData) thisModel.send('didResetTrackersData', req.didResetTrackersData)
         if (req.closePopup) window.close()
     })
 }
 
-const getBackgroundTabData = () => {
-    return new Promise((resolve, reject) => {
-        sendMessage('getCurrentTab').then((tab) => {
-            if (tab) {
-                sendMessage('getTab', tab.id).then((backgroundTabObj) => {
-                    resolve(backgroundTabObj)
-                })
-            }
-        })
-    })
+/** @typedef {ReturnType<import('../../background/message-handlers.js').getTab>} Tab */
+
+/**
+ * @returns {Promise<Tab|undefined>}
+ */
+async function getBackgroundTabData () {
+    const url = new URL(window.location.href)
+    let tabId
+    // Used for debugging only
```

Comment:
Would you mind to expand this comment a little bit, to explain how it's used for debugging? E.g. opening popup UI in a new tab and then setting the tab ID in URL manually.